src/client/lazy-app/Compress/index.tsx

```diff
@@ -374,6 +376,14 @@ export default class Compress extends Component<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props, prevState: State): void {
+    const wasLoading = prevState.loading;
+    const isLoading = this.state.loading || this.state.sides[0].loading || this.state.sides[1].loading;
```

Comment:
`this.state` and `prevState` are the same type, so you need to check all three loading points on `prevState` too.